model/map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/IckleQueryWhereClauses.java

```diff
@@ -167,6 +170,30 @@ private static String whereClauseForConsentClientFederationLink(String modelFiel
         return IckleQueryOperators.combineExpressions(ModelCriteriaBuilder.Operator.LIKE, getFieldName(UserModel.SearchableFields.CONSENT_FOR_CLIENT), new String[] {providerId + "%"}, parameters);
     }
 
+    private static String whereClauseForClientRole(String modelFieldName, ModelCriteriaBuilder.Operator op, Object[] values, Map<String, Object> parameters) {
+        if (op != ModelCriteriaBuilder.Operator.EQ && op != ModelCriteriaBuilder.Operator.NE) {
+            throw new CriterionNotSupportedException(RoleModel.SearchableFields.IS_CLIENT_ROLE, op);
+        }
+        if (values == null || values.length != 1) {
+            throw new CriterionNotSupportedException(RoleModel.SearchableFields.IS_CLIENT_ROLE, op, "Invalid arguments, expected (boolean), got: " + Arrays.toString(values));
+        }
+
+        Boolean b = (Boolean) values[0];
+        if (op == Operator.EQ) {
```

Comment:
This can be simplified to something like this:
```java
Boolean b = (Boolean) values[0];
        Operator operatorToUse = (op == Operator.EQ && b == Boolean.TRUE) || (op == Operator.NE && b == Boolean.FALSE) ? Operator.EXISTS : Operator.NOT_EXISTS;
        return IckleQueryWhereClauses.produceWhereClause(RoleModel.SearchableFields.CLIENT_ID, operatorToUse, new Object[] {}, parameters);
```

however, I am not sure about readability. Your version is probably better.